yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -340,6 +340,14 @@ private[spark] class Client(
           "for alternatives.")
     }
 
+    // Uploading $SPARK_CONF_DIR/log4j.properties file to the distributed cache to make sure that
+    // the executors will use the latest configurations instead of the default values. This is
+    // required when user changes log4j.properties directly to set the log configurations. If
+    // configuration file is provided through --files then executors will be taking configurations
+    // from --files instead of $SPARK_CONF_DIR/log4j.properties.
+    val log4jConf = oldLog4jConf.orElse(
+      Option(Utils.getContextOrSparkClassLoader.getResource("/log4j.properties")).map(_.toString))
```

Comment:
I might be over-estimating the intended scope, but this is not the only place that log4j.properties can occur. Generally callers can set -Dlog4j.configuration=... to set its location.